astroid/brain/brain_pathlib.py

```diff
@@ -0,0 +1,51 @@
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
+# For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from collections.abc import Iterator
+
+from astroid import bases, context, extract_node, inference_tip, nodes
+from astroid.exceptions import InferenceError, UseInferenceDefault
+from astroid.manager import AstroidManager
+
+PATH = extract_node(
+    """from pathlib import Path
+Path
+"""
+)
+PATH_CLASSDEF: nodes.ClassDef = next(PATH.infer())
```

Comment:
Inference calls are non-trivial. I'm don't think it's a good idea to do them at import. Better to only define the code template as global const and call `extract_node` (or even `_extract_single_node`) in `infer_parents_subscript`.

Reply:
But it isn't it better to use the actual `ClassDef`? You're more experienced here, but this seems to pass tests and doesn't this give us more information than a constructed mimic of `Path`?

Reply:
It's probably fine to use `extract_node` and `infer` to get the `ClassDef`, just don't do it at Module level / import time.

As example, this is how the `typing` brain does something similar
https://github.com/PyCQA/astroid/blob/e57ac42bca80c105bbdad4bae61637b39288a47b/astroid/brain/brain_typing.py#L99-L103
https://github.com/PyCQA/astroid/blob/e57ac42bca80c105bbdad4bae61637b39288a47b/astroid/brain/brain_typing.py#L167-L168

Reply:
I'll refactor!